runners/flink/src/test/java/org/apache/beam/runners/flink/PortableExecutionTest.java

```diff
@@ -0,0 +1,234 @@
+import org.apache.beam.runners.fnexecution.environment.EnvironmentFactory;
+import org.apache.beam.runners.fnexecution.environment.RemoteEnvironment;
+import org.apache.beam.runners.fnexecution.logging.GrpcLoggingService;
+import org.apache.beam.runners.fnexecution.provisioning.JobInfo;
+import org.apache.beam.runners.fnexecution.provisioning.StaticGrpcProvisionService;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.coders.BigEndianLongCoder;
+import org.apache.beam.sdk.coders.KvCoder;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.fn.IdGenerator;
+import org.apache.beam.sdk.fn.channel.ManagedChannelFactory;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.GroupByKey;
+import org.apache.beam.sdk.transforms.Impulse;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.WithKeys;
+import org.apache.beam.sdk.values.KV;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+/**
+ * Tests the execution of a pipeline from specification to execution on the portable Flink runner.
+ * Exercises job invocation, executable stage translation and deployment with embedded Flink for
+ * batch and streaming.
+ */
+@RunWith(Parameterized.class)
+public class PortableExecutionTest implements Serializable {
+
+  @Parameters
+  public static Object[] data() {
+    return new Object[] {true, false};
+  }
+
+  @Parameter public boolean isStreaming;
+
+  private transient ExecutorService sdkHarnessExecutor;
+  private transient Future<?> sdkHarnessExecutorFuture;
+
+  private DockerJobBundleFactory createJobBundleFactory(JobInfo jobInfo) throws Exception {
+    return new DockerJobBundleFactory(jobInfo) {
+
+      @Override
+      protected ServerFactory getServerFactory() {
+        return ServerFactory.createWithUrlFactory(
+            (host, port) -> HostAndPort.fromParts("localhost", port).toString());
+        //return InProcessServerFactory.create();
+      }
+
+      @Override
+      protected EnvironmentFactory getEnvironmentFactory(
+          GrpcFnServer<FnApiControlClientPoolService> controlServer,
+          GrpcFnServer<GrpcLoggingService> loggingServer,
+          GrpcFnServer<ArtifactRetrievalService> retrievalServer,
+          GrpcFnServer<StaticGrpcProvisionService> provisioningServiceServer,
+          Source clientSource,
+          IdGenerator idGenerator) {
+        String workerId = "fakeWorkerId";
+        ThreadFactory threadFactory = new ThreadFactoryBuilder().setDaemon(true).build();
+        // Create the SDK harness, and wait until it connects
+        sdkHarnessExecutor = Executors.newSingleThreadExecutor(threadFactory);
```

Comment:
We are currently assuming that there is a one to one relationship between environment creation and number of test runs.

Create the `ExecutorService` in `@Setup`.